File: mobile/lib/models/exercise.dart
class Exercise {
  final int id;
  final String name;
  final String description;
  final int sets;
  final int reps;
  final int restSeconds;

  Exercise({
    required this.id,
    required this.name,
    required this.description,
    required this.sets,
    required this.reps,
    required this.restSeconds,
  });

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      sets: json['sets'] ?? 0,
      reps: json['reps'] ?? 0,
      restSeconds: json['restSeconds'] ?? 0,
    );
  }
}

File: backend/Models/Exercise.cs
namespace backend.Models;

public class Exercise
{
    public int Id { get; set; }

    public int WorkoutPlanId { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public int Sets { get; set; }

    public int Reps { get; set; }

    public int RestSeconds { get; set; }

    // Explicit order inside the plan (1 = first).
    public int Position { get; set; }

    public WorkoutPlan WorkoutPlan { get; set; } = null!;
}
